wp-cache-export.php

```diff
@@ -0,0 +1,472 @@
+        $options[ $key ] = get_option( $key );
+        $validator->sanitize( $key, $value, $insert = true );
+      }
+      update_option( '_wp_super_cache_backup_options', $options );
+      unset( $settings[ '_wp_super_cache_options' ] );
+    }
+
+    // Create a new config file from the original sample
+    wp_cache_verify_config_file();
+    foreach ( $settings as $setting => $value) {
+      $validator->sanitize( $setting, $value, $insert = true );
+    }
+    include self::$cache_config_file;
+    // Reset the garbage collection to the new values
+    if ( function_exists( 'schedule_wp_gc' ) )
+      schedule_wp_gc( $forced = 1 );
+    wp_safe_redirect( add_query_arg( 'message', 0, $location ) );
+    exit;
+  }
+
+  /**
+   * Exports the current settings into a JSON file that can be imported into another WP Super Cache.
+   *
+   * When a user clicks the export button WordPress checks if the request is valid.
+   * Within the scope of this function, when the configuration file is included
+   * its variables are the only ones that are declared and gathered.  The additional options are
+   * then retrieved from the database. These variables are then encoded and downloaded
+   * as a JSON file.
+   *
+   * @since  1.4.4
+   *
+   * @uses  check_admin_referer
+   * @uses  nocache_header()
+   *
+   * @return JSON file of the WP Super Cache settings
+   */
+  public function export() {
+    if ( ! $this->can_export() || 0 !== count( get_defined_vars() ) ) {
+      return;
+    }
+    check_admin_referer(  self::EXPORT_NONCE );
+    include self::$cache_config_file;
+    $wp_cache_config_vars = get_defined_vars();
+    $wp_cache_config_options = $this->gather_plugin_options();
+    nocache_headers();
+    header( "Content-disposition: attachment; filename=" . self::TITLE );
+    header( 'Content-Type: application/octet-stream;' );
+    echo json_encode( array_merge($wp_cache_config_vars, $wp_cache_config_options )  );
+    die();
+  }
+
+  /**
+   * Restore the settings that existed before new settings were imported.
+   *
+   * This function will remove the current wp-cache-config.php file and replace it with the backup generated
+   * when settings were imported. The previous settings in the database are also reverted. The database option
+   * that stored these settings and their values is deleted once the restore is complete.
+   *
+   * @since  1.4.4
+   *
+   * @uses  check_admin_referer
+   * @uses  wp_safe_redirect
+   *
+   * @return  nothing Doesn't return anything. Redirects the user to the import/export form when the restore is complete.
+   *
+   */
+
+  public function restore() {
+    if ( ! $this->can_restore() ) {
+      return;
+    }
+    check_admin_referer( self::RESTORE_NONCE );
+    $location = add_query_arg( 'tab', 'export', admin_url( 'options-general.php?page=wpsupercache' ) );
+    if ( file_exists(self::$cache_config_file) )
+      @unlink( self::$cache_config_file );
```

Comment:
It isn't, the rename will overwrite it anyways. I put it there initially for clarity. Removed b47f3280